### PR TITLE
Create Scala documentation pages

### PR DESCRIPTION
**Pull request by gaeljw**
Now that [Cucumber Scala](https://github.com/cucumber/cucumber-jvm-scala) is to update, this PR aims to bring documentation about it in the official Cucumber docs.

It's basically adding Scala code samples in addition to Java and Kotlin ones.

### CODE REVIEW

**Comment by gaeljw** [comment]
I notice two things to fix in the preview:
- [x] the Scala SVG doesn't fit correctly, it stays at it's original size, probably something wrong with the SVG
- [x] the blocks of text for Scala are not visible, I guess there is something to activate somewhere but couldn't find where

**Comment by mpkorstanje** [comment]
@mlvandijk you added Kotlins recently, could you help out with the Hugo tags?

@gaeljw you may want to check Marits commits for Kotlin and see if you can find the magic thingy.

**Comment by gaeljw** [comment]
The 2 issues I mentionned are now solved.

I used another SVG file which resize properly, and added missing stuff in the `site.js` file to make the Scala tab works.

File: content/docs/cucumber/api.md [hunk 1]
```diff
@@ -50,6 +51,30 @@ In this case, the `DataTable` is automatically flattened to a `List<String>`
 by Cucumber (using `DataTable.asList(String.class)`) before invoking the step definition.
 {{% /text %}}
 
+{{% text "scala" %}}
+The simplest way to pass a `java.util.List[String]` to a step definition is to use a data table:
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]: This text is duplicated from the java section above. Could you merge the sections for java,kotlin,scala? Only the code examples are scala specific I think?

File: content/docs/cucumber/api.md [hunk 2]
```diff
@@ -65,13 +90,13 @@ Given I have 93 cucumbers in my belly
 
 In this step, you're "calling" the above step definition with one argument: the value `93`.
 
-Steps are declared in your {{% text "ruby" %}}`features/\*.feature`{{% /text %}}{{% text "java" %}}`*.feature`{{% /text %}}{{% text "javascript" %}}`*.feature`{{% /text %}} files.
+Steps are declared in your {{% text "ruby" %}}`features/\*.feature`{{% /text %}}{{% text "java" %}}`*.feature`{{% /text %}}{{% text "scala" %}}`*.feature`{{% /text %}}{{% text "javascript" %}}`*.feature`{{% /text %}} files.
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
Steps are declared in your {{% text "ruby" %}}`features/\*.feature`{{% /text %}}{{% text "java,kotlin,scala" %}}`*.feature`{{% /text %}}{{% text "javascript" %}}`*.feature`{{% /text %}} files.
```

Comment (mlvandijk) [reply content/docs/cucumber/api.md]: This should be how to make a text visible for multiple languages (i.e. things valid for all jvm languages)

File: content/docs/cucumber/api.md [hunk 3]
```diff
@@ -65,13 +90,13 @@ Given I have 93 cucumbers in my belly
...
 
 ## Matching steps
 
 1. Cucumber matches a step against a step definition's `Regexp`
 2. Cucumber gathers any capture groups or variables
-3. Cucumber passes them to the step definition's {{% text "ruby" %}}`Proc` (or “function”){{% /text %}}{{% text "javascript" %}}function{{% /text %}}{{% text "java" %}}method{{% /text %}} and executes it
+3. Cucumber passes them to the step definition's {{% text "ruby" %}}`Proc` (or “function”){{% /text %}}{{% text "javascript" %}}function{{% /text %}}{{% text "java" %}}method{{% /text %}}{{% text "scala" %}}method{{% /text %}} and executes it
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
3. Cucumber passes them to the step definition's {{% text "ruby" %}}`Proc` (or “function”){{% /text %}}{{% text "javascript" %}}function{{% /text %}}{{% text "java,scala" %}}method{{% /text %}} and executes it
```

File: content/docs/cucumber/api.md [hunk 4]
```diff
@@ -103,7 +128,7 @@ When a step definition's method or function invokes the `pending` method, the st
 
 When a step definition's method or function is executed and raises an error, the step is marked as failed (red). What you return from a step definition has no significance whatsoever.
 
-Returning {{% text "ruby" %}}`nil`{{% /text %}}{{% text "java" %}}`null`{{% /text %}}{{% text "javascript" %}}`null`{{% /text %}} or `false` will **not** cause a step definition to fail.
+Returning {{% text "ruby" %}}`nil`{{% /text %}}{{% text "java" %}}`null`{{% /text %}}{{% text "scala" %}}`null`{{% /text %}}{{% text "javascript" %}}`null`{{% /text %}} or `false` will **not** cause a step definition to fail.
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]: Same here

File: content/docs/cucumber/api.md [hunk 5]
```diff
@@ -114,6 +139,7 @@ Steps that follow `undefined`, `pending`, or `failed` steps are never executed,
 Step definitions have to be unique for Cucumber to know what to execute.
 If you use ambiguous step definitions,{{% text "ruby" %}}Cucumber will raise a `Cucumber::Ambiguous` error,{{% /text %}}
 {{% text "java" %}} Cucumber will raise an `AmbiguousStepDefinitionsException`,{{% /text %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]: Please merge java,kotlin,scala (as above)

File: content/docs/cucumber/api.md [hunk 6]
```diff
@@ -129,6 +155,10 @@ If you want more fine-grained control, you can use [conditional hooks](#conditio
 You can declare hooks in any class.
 {{% /text %}}
 
+{{% text "scala" %}}
+You can declare hooks in any class, trait or object.
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md:148]: I wonder whether this is now shown twice for scala; i.e. both "You can declare hooks in any class." and "You can declare hooks in any class, trait or object." - if so please adapt the sentence to show "class" for some languages and "class, trait or object." for scala only.

File: content/docs/cucumber/api.md [hunk 7]
```diff
@@ -354,11 +409,11 @@ Around('@fast') do |scenario, block|
 end
 ```
 
-{{% block "java" %}}Cucumber-JVM does not support `Around` hooks.{{% /block %}}
+{{% block "java,scala" %}}Cucumber-JVM does not support `Around` hooks.{{% /block %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
{{% block "java,kotlin,scala" %}}Cucumber-JVM does not support `Around` hooks.{{% /block %}}
```

Comment (mlvandijk) [reply content/docs/cucumber/api.md]: Hijacking your PR a little to make some Kotlin improvements too if you don't mind ;)

File: content/docs/cucumber/api.md [hunk 8]
```diff
@@ -354,11 +409,11 @@ Around('@fast') do |scenario, block|
...
 {{% block "javascript" %}}Cucumber.js does not support `Around` hooks.{{% /block %}}
 
 ## Step hooks
-{{% text "java" %}}
+{{% text "java,scala" %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
{{% text "java,kotlin,scala" %}}
```

File: content/docs/cucumber/api.md [hunk 9]
```diff
@@ -432,13 +495,23 @@ AfterStep { scenario: Scenario ->
...
+
+{{% /block %}}
+
 {{% block "javascript" %}}Cucumber.js does not support `AfterStep` hooks.{{% /block %}}
 
 ## Conditional hooks
 
 Hooks can be conditionally selected for execution based on the tags of the scenario.
 To run a particular hook only for certain scenarios, you can associate a
 {{% text "java" %}}`Before` or `After`{{% /text %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
{{% text "java,kotlin,scala" %}}`Before` or `After`{{% /text %}}
```

File: content/docs/cucumber/api.md [hunk 10]
```diff
@@ -432,13 +495,23 @@ AfterStep { scenario: Scenario ->
...
+{{% text "scala" %}}`Before` or `After`{{% /text %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
```

Comment (mlvandijk) [reply content/docs/cucumber/api.md]: Please merge ;)

File: content/docs/cucumber/api.md [hunk 11]
```diff
@@ -506,7 +587,7 @@ at_exit do
 end
 ```
 
-{{% block "java" %}}Cucumber-JVM does not support global hooks.{{% /block %}}
+{{% block "java,scala" %}}Cucumber-JVM does not support global hooks.{{% /block %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
{{% block "java,kotlin,scala" %}}Cucumber-JVM does not support global hooks.{{% /block %}}
```

File: content/docs/cucumber/api.md [hunk 12]
```diff
@@ -525,7 +606,7 @@ end
 ```
 {{% /text %}}
 
-{{% block "java" %}}Cucumber-JVM does not support running a hook only once.{{% /block %}}
+{{% block "java,scala" %}}Cucumber-JVM does not support running a hook only once.{{% /block %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
{{% block "java,kotlin,scala" %}}Cucumber-JVM does not support running a hook only once.{{% /block %}}
```

File: content/docs/cucumber/api.md [hunk 13]
```diff
@@ -548,7 +629,7 @@ This hook will run _only once_: after support has been loaded, and before any fe
 You can use this hook to extend Cucumber. For example you could affect how features are loaded, or register custom formatters programmatically.
 {{% /text %}}
 
-{{% text "java" %}}Cucumber-JVM does not support `AfterConfiguration` hooks.{{% /text %}}
+{{% text "java,scala" %}}Cucumber-JVM does not support `AfterConfiguration` hooks.{{% /text %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
{{% text "java,kotlin,scala" %}}Cucumber-JVM does not support `AfterConfiguration` hooks.{{% /text %}}
```

File: content/docs/cucumber/api.md [hunk 14]
```diff
@@ -804,19 +899,21 @@ The most common option is to run Cucumber from the command line.
 
 By default, Cucumber will treat anything ending in
 {{% text "java" %}}`.java`{{% /text %}}
+{{% text "scala" %}}`.scala`{{% /text %}}
 {{% text "kotlin" %}}`.kt`{{% /text %}}
 {{% text "javascript" %}}`.js`{{% /text %}}
 {{% text "ruby" %}}`.rb`{{% /text %}} under the root
-{{% text "java,kotlin,javascript" %}}resource{{% /text %}}
+{{% text "java,kotlin,javascript,scala" %}}resource{{% /text %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]: I'd prefer scala with the jvm languages "java,kotlin,scala,javascript" but that might be a personal preference.

File: content/docs/cucumber/api.md [hunk 15]
```diff
@@ -804,19 +899,21 @@ The most common option is to run Cucumber from the command line.
...
 
 Thus, a step contained in
 {{% text "java" %}}`features/models/entities/step-definitions/anything.java`{{% /text %}}
 {{% text "kotlin" %}}`features/models/entities/step-definitions/anything.kt`{{% /text %}}
+{{% text "scala" %}}`features/models/entities/step-definitions/anything.scala`{{% /text %}}
 {{% text "javascript" %}}`features/models/entities/step-definitions/anything.js`{{% /text %}}
 {{% text "ruby" %}}`features/models/entities/step_definitions/anything.rb`{{% /text %}}
 can be used in a feature file contained in
-{{% text "java,kotlin,javascript" %}}`features/views/entity-new`{{% /text %}}
+{{% text "java,kotlin,javascript,scala" %}}`features/views/entity-new`{{% /text %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]: Same here

File: content/docs/cucumber/api.md [hunk 16]
```diff
@@ -1238,7 +1447,7 @@ cucumber --help
 ```
 {{% /block %}}
 
-{{% block "java" %}}
+{{% block "java,scala" %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
{{% block "java,kotlin,scala" %}}
```

File: content/docs/cucumber/api.md [hunk 17]
```diff
@@ -1253,7 +1462,7 @@ Use the `cucumber-js --help` command to see which arguments can be passed to the
 
 You can also use [tags](#tags) to specify what to run.
 
-{{% block "java" %}}
+{{% block "java,scala" %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/api.md]:
```suggestion
{{% block "java,kotlin,scala" %}}
```

File: content/docs/cucumber/configuration.md [hunk 18]
```diff
@@ -8,20 +8,21 @@ polyglot:
...
-{{% block "java,kotlin" %}}
+{{% block "java,kotlin,scala" %}}
 Parameter types let you convert parameters from cucumber-expressions to objects.
 Data table and doc string types let you convert data tables and doc
 strings to objects. Like step definitions, type definitions are part of the glue.
 When placed on the glue path Cucumber will detect them automatically. 
 {{% /block %}}
 
-{{% block "java" %}}
+{{% block "java,scala" %}}
```
Comment (mlvandijk) [inline content/docs/cucumber/configuration.md]:
```suggestion
{{% block "java,kotlin,scala" %}}
```

File: content/docs/installation/scala.md [hunk 19]
```diff
@@ -1,11 +1,51 @@
...
+Make sure the Cucumber version is the same for all Cucumber dependencies.
+{{% /note %}}
+
+# Maven
+
+```xml
+<dependency>
+    <groupId>io.cucumber</groupId>
+    <artifactId>cucumber-scala_2.13</artifactId>
+    <version>{{% version "cucumberjvm" %}}</version>
```
Comment (mlvandijk) [inline content/docs/installation/scala.md]: Will the scala version always match the "cucumberjvm" version? If not, I suggest adding its own version number. In versions.yaml I think.

File: content/docs/installation/scala.md [hunk 20]
```diff
@@ -1,11 +1,51 @@
...
+    <scope>test</scope>
+</dependency>
+```
+
+You can now run Cucumber [from the command line](/docs/cucumber/api/#from-the-command-line) or [run Cucumber with Maven](/docs/tools/java#maven).
+
+# Sbt
```
Comment (mlvandijk) [inline content/docs/installation/scala.md:37]: What is "Sbt"? Or would everyone using Scala know?

**Review by mlvandijk — CHANGES_REQUESTED** [review]
Hi @gaeljw - this is amazing!! Thanks so much for taking the time to do this. <3

I do have some comments/suggestions, most are very small and you should be able to just accept my suggestions. I'm most worried about the version, and would prefer to add a separate version number for scala if it is a different repo / release cycle. 
Also, I have not been able to check your Scala code, so if you know anyone who could check that, that would be great.

File: content/docs/cucumber/api.md (hunk 1 above)
Comment (gaeljw) [reply content/docs/cucumber/api.md]: There is a slight difference in the way we define generic types between Java and Scala : `List<String>` vs `List[String]`. But still we can merge some stuff together.

File: content/docs/cucumber/api.md (hunk 2 above)
Comment (gaeljw) [reply content/docs/cucumber/api.md]: When writing first pages I didn't notice that we can use multiple languages at the same time.. sorry!

File: content/docs/installation/scala.md (hunk 19 above)
Comment (gaeljw) [reply content/docs/installation/scala.md]: There's a small chance/risk that version will not always match indeed!

File: content/docs/installation/scala.md (hunk 20 above)
Comment (gaeljw) [reply content/docs/installation/scala.md:37]: It's one of the most used build tool for Scala (roughly replaces Maven or Gradle). I think everyone who uses Scala knows about sbt.

File: content/docs/cucumber/api.md (hunk 6 above)
Comment (gaeljw) [reply content/docs/cucumber/api.md:148]: It looks fine in the preview :)

**Comment by gaeljw** [comment]
Thanks a lot for the review @mlvandijk 

I've applied all of your comments :)

The code samples are taken from the Cucumber Scala doc itself, which is itself mostly a copy of the tests we have in the Cucumber Scala project. A bit too much copy/paste but I'm quite confident with the code samples.

File: content/docs/cucumber/api.md (hunk 1 above)
Comment (mlvandijk) [reply content/docs/cucumber/api.md]: Yes, you can also use most of the sentence (the text that is the same) for both and only distinguish between the parts that are different - like we do in other places

File: content/docs/cucumber/api.md (hunk 2 above)
Comment (mlvandijk) [reply content/docs/cucumber/api.md]: No problem. :)

File: content/docs/cucumber/api.md (hunk 6 above)
Comment (mlvandijk) [reply content/docs/cucumber/api.md:148]: It does. thnx

**Review by mlvandijk — APPROVED** [review]

**Comment by mlvandijk** [comment]
Thanks so much for this @gaeljw !

**Pull request merged**